pkg/controller/masteruserrecord/masteruserrecord_controller_test.go

```diff
@@ -0,0 +1,200 @@
+	"testing"
+)
+
+const (
+	hostClusterName   = "host-cluster"
+	memberOperatorNs  = "toolchain-member-operator"
+	memberClusterName = "member-cluster"
+	hostOperatorNs    = "toolchain-host-operator"
+)
+
+type getMemberCluster func(cl client.Client) func(name string) (*cluster.FedCluster, bool)
+
+func TestCreateUserAccountSuccessful(t *testing.T) {
+	// given
+	logf.SetLogger(logf.ZapLogger(true))
+	s := apiScheme(t)
+	mur := newMasterUserRecord("john")
+	memberClient := fake.NewFakeClientWithScheme(s)
+	hostClient := fake.NewFakeClientWithScheme(s, mur)
+	cntrl := newController(hostClient, memberClient, s, newGetMemberCluster(true, v1.ConditionTrue))
+
+	// when
+	_, err := cntrl.Reconcile(newMurRequest(mur))
+
+	// then
+	require.NoError(t, err)
+	ua := &toolchainv1alpha1.UserAccount{}
+	err = memberClient.Get(context.TODO(), namespacedName(memberOperatorNs, "john"), ua)
+	require.NoError(t, err)
+	assert.EqualValues(t, mur.Spec.UserAccounts[0].Spec, ua.Spec)
+	assertMurCondition(t, hostClient, toBeNotReady(provisioningReason, ""))
+}
+
+func TestCreateUserAccountFailed(t *testing.T) {
+	// given
+	logf.SetLogger(logf.ZapLogger(true))
+	s := apiScheme(t)
+	mur := newMasterUserRecord("john")
+	memberClient := fake.NewFakeClientWithScheme(s)
+	hostClient := fake.NewFakeClientWithScheme(s, mur)
+
+	t.Run("when member cluster does not exist", func(t *testing.T) {
+		// given
+		cntrl := newController(hostClient, memberClient, s, newGetMemberCluster(false, v1.ConditionTrue))
+
+		// when
+		_, err := cntrl.Reconcile(newMurRequest(mur))
+
+		// then
+		require.Error(t, err)
+		msg := "the fedCluster member-cluster not found in the registry"
+		assert.Contains(t, err.Error(), msg)
+		assertUaNotFound(t, memberClient)
+		assertMurCondition(t, hostClient, toBeNotReady(clusterNotReady, msg))
+	})
+
+	t.Run("when member cluster does not exist", func(t *testing.T) {
+		// given
+		cntrl := newController(hostClient, memberClient, s, newGetMemberCluster(true, v1.ConditionFalse))
+
+		// when
+		_, err := cntrl.Reconcile(newMurRequest(mur))
+
+		// then
+		require.Error(t, err)
+		msg := "the fedCluster member-cluster is not ready"
+		assert.Contains(t, err.Error(), msg)
+		assertUaNotFound(t, memberClient)
+		assertMurCondition(t, hostClient, toBeNotReady(clusterNotReady, msg))
+	})
+}
+
+func assertMurCondition(t *testing.T, hostClient client.Client, expected toolchainv1alpha1.Condition) {
+	record := &toolchainv1alpha1.MasterUserRecord{}
+	err := hostClient.Get(context.TODO(), namespacedName(hostOperatorNs, "john"), record)
+	require.NoError(t, err)
+	test.AssertConditionsMatch(t, record.Status.Conditions, expected)
+}
+
+func assertUaNotFound(t *testing.T, memberClient client.Client) {
+	ua := &toolchainv1alpha1.UserAccount{}
+	err := memberClient.Get(context.TODO(), namespacedName(memberOperatorNs, "john"), ua)
+	require.Error(t, err)
+	assert.IsType(t, metav1.StatusReasonNotFound, errors.ReasonForError(err))
+}
+
```

Comment:
Let's also add a test(s) for the case when the status update failed (something like https://github.com/codeready-toolchain/member-operator/blob/master/pkg/controller/useraccount/useraccount_controller_test.go#L414)

Reply:
done c0939f4